internal/api/plugin/migrate/migrate.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2025 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migrate
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/build"
+	"cuelang.org/go/cue/cuecontext"
+	apiinterface "github.com/perses/perses/internal/api/interface"
+	"github.com/perses/perses/internal/api/plugin/schema"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/perses/perses/pkg/model/api/v1/common"
+	"github.com/perses/perses/pkg/model/api/v1/dashboard"
+	"github.com/perses/perses/pkg/model/api/v1/plugin"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	grafanaType     = "#grafanaType"
+	migrationFolder = "migrate"
+)
+
+func LoadMigrateSchema(schemaPath string) (*build.Instance, error) {
+	return schema.LoadSchemaInstance(schemaPath, "migrate")
+}
+
+func ReplaceInputValue(input map[string]string, grafanaDashboard string) string {
+	result := grafanaDashboard
+	for key, value := range input {
+		// Escape special characters that may be present in the value
+		escapedValue := strconv.Quote(value)
+		// Remove the extra surrounding quotes added by strconv.Quote
+		escapedValue = escapedValue[1 : len(escapedValue)-1]
+		// Do the replacement (2 syntaxes to support)
+		result = strings.ReplaceAll(result, fmt.Sprintf("$%s", key), escapedValue)
+		result = strings.ReplaceAll(result, fmt.Sprintf("${%s}", key), escapedValue)
+	}
+	return result
+}
+
+// TODO Put in common with percli plugin lint command.
+func isPackageMigrate(file string) (bool, error) {
+	data, err := os.ReadFile(file)
+	if err != nil {
+		return false, err
+	}
+	return strings.Contains(string(data), "package migrate"), nil
+}
+
```

Comment:
maybe a comment to explain the "guessing" logic of this function?

Reply:
my remark was actually for getPluginKind 😄

Reply:
ah. Well I will put a comment when coming back to this part, which will be soon enough to not forget haha